pylana/utils.py

```diff
@@ -5,13 +5,97 @@
 from collections import defaultdict
 from typing import Iterable, Dict, List
 
+import re
 import pandas as pd
 
+def create_aggregation_function(aggregation_function: str) -> dict:
+    """
+    Create a dictionary containing the aggregation function in a format necessary
+    for the aggregation API request.
+
+    Args:
+        aggregation_function:
+            A string denoting the aggregation function to use.
+            Can be one of ["min", "max", "sum", "mean", "median", "variance",
+            "standardDeviation"] or a percentile as a string starting with "p"
+            followed by the percentile value.
+
+    Returns:
+        A dictionary containing the aggregation function string.
+    """
+    if re.match('^p[0-9]{1,3}', aggregation_function):
+        return({'aggregationFunction': {'type': 'percentile',
+                                        'percentile': int(aggregation_function[1:])}})
+    else:
+        return({'aggregationFunction': aggregation_function})
+
+
+def create_metric(metric_value: str, aggregation_function: str = 'sum') -> dict:
```

Comment:
I think @bosyk solution is certainly more robust but I think @jonaslana original solution definitely reads more simple and is less "branchy". Perhaps there is a good middle ground?

Reply:
I guess we accepted my suggestion. I hope it will survive the test of time in form of extensions of the api. If they result in writing new code rather than modifying several functions, it does what it is supposed to do.